packages/@sanity/core/src/commands/dataset/alias/unlinkAliasHandler.js

```diff
@@ -0,0 +1,59 @@
+import promptForDatasetAliasName from '../../../actions/dataset/alias/datasetAliasNamePrompt'
+import validateDatasetAliasName from '../../../actions/dataset/alias/validateDatasetAliasName'
+import * as aliasClient from './datasetAliasesClient'
+import {ALIAS_PREFIX} from './datasetAliasesClient'
+
+export default async (args, context) => {
+  const {apiClient, output, prompt} = context
+  const [, alias] = args.argsWithoutOptions
+  const client = apiClient()
+
+  const nameError = alias && validateDatasetAliasName(alias)
+  if (nameError) {
+    throw new Error(nameError)
+  }
+
+  const [fetchedAliases] = await Promise.all([aliasClient.listAliases(client)])
+  const aliases = fetchedAliases.map((da) => da.name)
+
+  let aliasName = await (alias || promptForDatasetAliasName(prompt))
+  let aliasOutputName = aliasName
+
+  if (aliasName.startsWith(ALIAS_PREFIX)) {
+    aliasName = aliasName.substring(1)
+  } else {
+    aliasOutputName = `${ALIAS_PREFIX}${aliasName}`
+  }
+
+  if (!aliases.includes(aliasName)) {
+    throw new Error(`Dataset alias "${aliasOutputName}" does not exist `)
+  }
+
+  // get the current alias from the remote alias list
+  const linkedAlias = fetchedAliases.find((elem) => elem.name === aliasName)
+
+  if (linkedAlias) {
```

Comment:
What happens if `.find` doesn't find anything? 🙂

Reply:
The intention is probably to throw the error below, but I think the code needs to be moved.

Reply:
Why do you think the intention is to throw an error. Please go through the logic again and understand what its doing :)

Reply:
It `.find` doesn't find anything given the predicate, it returns undefined.

So shouldn't the `if (!linkedAlias.datasetName)` be moved to before this `if` and then throw the error?

Reply:
After re-reading this, as far as I can tell `linkedAlias` will actually always find a value, given the check on line 28, which means this `if` branch isn't actually necessary? I think it would read better if refactored so the check on line 28 uses `linkedAlias` here instead.